### DISCUSSION TITLE
Is it possible to avoid the "% Compiled module:" message?

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#1763] [Q&A]
Is it possible to avoid the "% Compiled module:" message for every compiled program?
I really do not need this info and the important print messages get hidden.

**Comment by GillesDuvert** [comment]
Certainly easy for GDL, a couple of code lines could suffice... But, is there some equivalent in IDL we should reproduce? Never saw anything about it in the documentation.

OTHOH, as GDL now can save procedures, and does not print the "% Compiled module" when restoring them, one could with some work put many of them in a `.sav` file and just restore the `.sav` at start of gdl. Additionnaly this saves a lot of, "compile time".

**Reply by brandy125** [reply]
@GillesDuvert you mentioned in your post that saving procedures in GDL is now available. How does it work?
I tested it but for me it does not work.
Do I have to use a special version?

**Comment by brandy125** [comment]
I would really appreciate if GDL could be silent on this. 

Didn't know that GDL can save and restore procedures! Since when? Will test it right now.

**Comment by alaingdl** [comment]
@brandy125 
It is possible to avoid those messages by adding in each func/pro a ligne : 
`COMPILE_OPT idl2, hidden`

In fact it is a problem since IDL is not enforcing that at the compilation level !

I suspect it is quite simple to add a global env to deactivate that. The code to look at is `dcompiler.cpp` (lines 255 & 322)

We have 
```
~/GDL/last-gdl/src$ grep -ni 'Compiled module' *
[...]
gdlc.g:179:        GDL_HIDDEN=256 //flag to avoid writing "Compiled module" at compilation. 
GDLParser.hpp:50:        GDL_HIDDEN=256 //flag to avoid writing "Compiled module" at compilation. 
```
but I don't know what it stand for :(

**Reply by GillesDuvert** [reply]
wow, no, no need to mess with gdl antlr parser!
I would just implement a global switch like --silent

**Reply by brandy125** [reply]
@alaingdl nice! Thanks!

**Comment by alaingdl** [comment]
You need to explicitly specify the /routine keyword

Tested with success on Linux (U2204) on current Git version (compiled with the script) 

```
First GDL session (working on ~) :

GDL> .r test_loops.pro
% Compiled module: TEST_LOOP_REPEAT.
% Compiled module: TEST_LOOP_WHILE.
% Compiled module: TEST_LOOP_FOR.
% Compiled module: TEST_LOOPS.
GDL> save, file='toto',/rout

```
then 

```
Second GDL session : 

GDL> restore, '~/toto'
GDL> help,/pro
Compiled Procedures:
$MAIN$
TEST_LOOPS               NBP 
TEST_LOOP_FOR            NBP 
TEST_LOOP_REPEAT         NBP 
TEST_LOOP_WHILE          NBP 
GDL> help,/pro,/ful
Compiled Procedures:
$MAIN$
TEST_LOOPS               NBP 
TEST_LOOP_FOR            NBP 
TEST_LOOP_FOREACH        NBP    <<-- compiled with hidden ... 
TEST_LOOP_REPEAT         NBP 
TEST_LOOP_WHILE          NBP 
GDL> TEST_LOOP_REPEAT
% Time elapsed : 0.0097720623 seconds.

```

You cannot save in same file variables & routines

We need few cleanups of the messages when doing unexpected calls ...

**Comment by brandy125** [comment]
Wow wow wow!!!!
Very nice!!!
Thank you guys!

Nice work done!

A full compilation of my 1400 programs took 96 seconds and a restore now takes only 3 seconds!!!
Very helpful!

**Reply by GillesDuvert** [reply]
Actually the idea is to provide in the near future all available procedures as already compiled, this will save a lot of time for everyone.
And probably caching already defined procedures as compiled in the .gdl cache also.
For the record, the procedures are not fully compiled, only the first stage is produced: what atomic commands the parser has created, but most of the compile time is devoted to this, and this is OS-independent. Only a change of the parser will call for an update of those semicompiled files, so they are versioned and work only with the 'good' parser.

**Comment by GillesDuvert** [comment]
added --silent in #1769

**Comment by GillesDuvert** [comment]
The definitive answer:
call gdl with option` --silent` . You need gdl >= 1.0.5 for this to work
